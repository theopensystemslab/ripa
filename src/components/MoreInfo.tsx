import Drawer from "@material-ui/core/Drawer";
import IconButton from "@material-ui/core/IconButton";
import { makeStyles } from "@material-ui/core/styles";
import CloseIcon from "@material-ui/icons/Close";
import classNames from "classnames";
import * as marked from "marked";
import * as React from "react";

const useStyles = makeStyles(theme => ({
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
    color: theme.palette.text.primary,
    transition: theme.transitions.create("margin", {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen
    }),
    marginRight: -drawerWidth,
    [theme.breakpoints.only("xs")]: {
      width: "100%",
      marginRight: "-100%"
    }
  },
  drawerShift: {
    marginRight: 0
  },
  drawerTitle: {
    marginTop: theme.spacing(1),
    fontWeight: 700,
    fontSize: "1.15rem",
    flexGrow: 1
  },
  drawerContent: {
    padding: "0.5rem 1.75rem 1rem",
    fontSize: "1rem",
    lineHeight: "1.5",
    "& a": {
      color: theme.palette.grey[500],
      "&:hover": {
        color: theme.palette.grey[700]
      }
    }
  },
  drawerPaper: {
    width: drawerWidth,
    [theme.breakpoints.only("xs")]: {
      width: "100%"
    },
    padding: theme.spacing(1)
  },
  close: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end"
  }
}));

const drawerWidth = 400;
const MoreInfo = ({ content, open = false, handleClose }) => {
  const html = { __html: marked(content, { sanitize: true }) };
  const classes = useStyles();
  return (
    <>
      <Drawer
        className={classNames(classes.drawer, {
          [classes.drawerShift]: open
        })}
        variant="persistent"
        anchor="right"
        open={open}
        classes={{
          paper: classes.drawerPaper
        }}
      >
        <div className={classes.close}>
          <IconButton onClick={handleClose} aria-label="Close Panel">
            <CloseIcon />
          </IconButton>
        </div>
        <div className={classes.drawerContent}>
          <div dangerouslySetInnerHTML={html} />
        </div>
      </Drawer>
    </>
  );
};

export default MoreInfo;
