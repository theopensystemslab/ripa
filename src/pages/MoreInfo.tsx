import * as marked from "marked";
import * as React from "react";

const MoreInfo = ({ content }) => {
  const html = { __html: marked(content, { sanitize: true }) };
  return (
    <>
      <div dangerouslySetInnerHTML={html} />
    </>
  );
};

export default MoreInfo;
