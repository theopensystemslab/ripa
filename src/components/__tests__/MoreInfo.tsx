import { render } from "@testing-library/react";
import * as React from "react";

import MoreInfo from "../MoreInfo";

const setOpen = jest.fn();

describe("More Info Component", () => {
  it("should render snapshot", () => {
    const { asFragment } = render(
      <MoreInfo
        open={false}
        handleClose={setOpen}
        content={`## Why it matters`}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
