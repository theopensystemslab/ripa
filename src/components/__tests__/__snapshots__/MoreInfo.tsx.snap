// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`More Info Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
    >
      <div
        class="makeStyles-close-6"
      >
        <button
          aria-label="Close Panel"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="closeBtn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="makeStyles-drawerContent-4"
      >
        <div>
          <h2
            id="why-it-matters"
          >
            Why it matters
          </h2>
          

        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
